### PR TITLE
[BACKLOG-36851] : Focus Visible - Accessibility -  PUC Home Page –Dat…

### PR DESCRIPTION
**Pull request by addagudi**
…a Source Wizard (SQL Query)

### CODE REVIEW

**Comment by addagudi** [comment]
@pentaho/wcag Please review

**Comment by buildguy** [comment]
### :x: Build failed in 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 15s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
:ok_hand: All tests passed!
 
Tests run: **18**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/xul/job/PR-228/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by hv-vamsi — APPROVED** [review]

**Review by phalguni-a — APPROVED** [review]

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java [hunk 1]
```diff
@@ -155,6 +154,9 @@ public void init( com.google.gwt.xml.client.Element srcEle, XulDomContainer cont
     if ( !StringUtils.isEmpty( srcEle.getAttribute( "onclick" ) ) ) {
       setOnclick( srcEle.getAttribute( "onclick" ) ); //$NON-NLS-1$
     }
+    if ( !StringUtils.isEmpty( srcEle.getAttribute( "tabindex" ) ) ) {
```
Comment (dcleao) [inline gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java:157]: So `tabindex` does exist as a standard XUL attribute (https://web.archive.org/web/20150921043125/https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XUL/Attribute/tabindex), and so using the `pen` namespace (prefix) is not necessary.

However, this attribute applies general to all XUL elements, as far as I know, or, at least, would not hurt and would be quite advantageous, if it were available to all elements.

As such, I believe that support for this attribute should instead be implemented in `AbstractGwtXulComponent`.

The pattern there followed to implement other XUL attributes should be followed (check the `init`, `setAttribute` methods, the `Property` enum).

For an example of a recently added attribute (albeit a "pen" one), check `ARIA_ROLE`, `getAriaRole`, `setAriaRole`).

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java [hunk 2]
```diff
@@ -203,6 +205,10 @@ public void setSrc( Object img ) {
     }
   }
 
+  public void setTabIndex( String tabIndex ) {
+    image.getElement().setTabIndex( Integer.parseInt( tabIndex ) );
```
Comment (dcleao) [inline gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java:209]: What if `tabIndex` is empty?

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 41s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
:ok_hand: All tests passed!
 
Tests run: **18**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/xul/job/PR-228/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java (hunk 1 above)
Comment (addagudi) [reply gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java]: Modified the approach as per your suggestion.

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java (hunk 2 above)
Comment (addagudi) [reply gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtImage.java]: Addressed

**Review by ddiroma — APPROVED** [review]

**Review by ddiroma — APPROVED** [review]

**Review by dcleao — CHANGES_REQUESTED** [review]
Today I met with @addagudi and requested some additional changes so that one of the existing patterns for XUL component attributes/properties is followed, and with that supporting more usage scenarios as I believe a generic XUL attribute such as `tabindex`, implemented in the base class, warrants.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 58s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,gwt/impl
```
---
:ok_hand: All tests passed!
 
Tests run: **371**, Failures: **0**, Skipped: **295** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/xul/job/PR-228/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,gwt/impl
```
---
:ok_hand: All tests passed!
 
Tests run: **371**, Failures: **0**, Skipped: **295** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/xul/job/PR-228/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 59s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,gwt/impl
```
---
:ok_hand: All tests passed!
 
Tests run: **371**, Failures: **0**, Skipped: **295** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/xul/job/PR-228/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,gwt/impl
```
---
:ok_hand: All tests passed!
 
Tests run: **371**, Failures: **0**, Skipped: **295** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/xul/job/PR-228/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
:ok_hand: All tests passed!
 
Tests run: **18**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/xul/job/PR-228/10/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Pull request merged**